src/julia_gc.c

```diff
@@ -531,7 +551,19 @@ void GapTaskScanner(jl_task_t * task, int root_task)
     int    tid;
     void * stack = jl_task_stack_buffer(task, &size, &tid);
     if (stack && tid < 0) {
-        TryMarkRange(stack, (char *)stack + size);
+        if (task->copy_stack) {
+            // We know the actually used part of the task stack, so
+            // shorten the range we have to scan.
+            stack = (void *)((char *) stack + size - task->copy_stack);
+            size = task->copy_stack;
+        }
```

Comment:
Please insert a brief comment here, perhaps like this:
> // Since we don't know the exact size of the stack, we might run into a read protected page while scanning it, triggering a segfault. Julia catches this, and longjmp's to `JuliaTLS->safe_restore`; we use this mechanism to abort the stack scanning as soon as we hit a protected page. For this to work, we must scan the stack from top to bottom.